### PR TITLE
create druid package

### PR DESCRIPTION
**Pull request by sanghanan**
Fixes:

Related: https://github.com/chainguard-dev/image-requests/issues/1612

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

**Comment by sanghanan** [comment]
This package has a lot of shaded JARs. JARs that we could bump needed upstream code changes. 😞 
Advisories: https://github.com/sanghanan/advisories/blob/druid/druid.advisories.yaml

File: druid.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      packages:
+        - python3
+        - perl
+  pipeline:
+    - runs: |
+        /usr/share/java/druid/bin/start-nano-quickstart > /usr/share/java/druid/logs.txt 2>&1 &
+        DRUID_PID=$!
+        sleep 2
+        grep -q "Starting Apache Druid" /usr/share/java/druid/logs.txt
+        kill $DRUID_PID
```
Comment (Dentrax) [inline druid.yaml:77]: It seems Druid [provides an API](https://druid.apache.org/docs/latest/api-reference/service-status-api/) to service status, should we do quick health check using `curl` just after `grep` tests?

FWIW, you can even use `test/daemon-check-output` here.

**Comment by Dentrax** [comment]
> This package has a lot of shaded JARs. JARs that we could bump needed upstream code changes. 😞 Advisories: https://github.com/sanghanan/advisories/blob/druid/druid.advisories.yaml

Great work here!

Any luck with `maven/pombump` pipeline? It may helpful for non-shaded ones.

File: druid.yaml (hunk 1 above)
Comment (sanghanan) [reply druid.yaml:77]: The entrypoint script that lets you use the API needs a cluster to work correctly. I have included it in the image PR though: https://github.com/chainguard-images/images-private/blob/f00a7b2f6b5382cffc5189f68024cb83b81d3760/images/druid/tests/test.sh#L12

Thanks for the tip about `test/daemon-check-output`, will look into that

**Comment by sanghanan** [comment]
> Any luck with `maven/pombump` pipeline? It may helpful for non-shaded ones.

Unfortunately, anything I could bump with pombump ended up breaking the build - it'll need a few code changes from the maintainers.

**Review by jamonation — APPROVED** [review]

**Pull request merged**